src/python/pants/option/arg_splitter_test.py

```diff
@@ -98,140 +86,259 @@ def assert_not_positional(arg):
-
-  def test_help_detection(self):
-    self._split_help('./pants', [], {'': []}, [])
-    self._split_help('./pants -f', [], {'': ['-f']}, [])
-    self._split_help('./pants help', [], {'': []}, [])
-    self._split_help('./pants -h', [], {'': []}, [])
-    self._split_help('./pants --help', [], {'': []}, [])
-    self._split_help('./pants help compile -x', ['compile'],
-                {'': [], 'compile': ['-x']}, [])
-    self._split_help('./pants help compile -x', ['compile'],
-                {'': [], 'compile': ['-x']}, [])
-    self._split_help('./pants compile -h', ['compile'],
-                {'': [], 'compile': []}, [])
-    self._split_help('./pants compile --help test', ['compile', 'test'],
-                {'': [], 'compile': [], 'test': []}, [])
-    self._split_help('./pants test src/foo/bar:baz -h', ['test'],
-                {'': [], 'test': []}, ['src/foo/bar:baz'])
-
-    self._split_help('./pants help-advanced', [], {'': []}, [], True, False)
-    self._split_help('./pants help-all', [], {'': []}, [], False, True)
-    self._split_help('./pants --help-advanced', [], {'': []}, [], True, False)
-    self._split_help('./pants --help-all', [], {'': []}, [], False, True)
-    self._split_help('./pants --help --help-advanced', [], {'': []}, [], True, False)
-    self._split_help('./pants --help-advanced --help', [], {'': []}, [], True, False)
-    self._split_help('./pants --help --help-all', [], {'': []}, [], False, True)
-    self._split_help('./pants --help-all --help --help-advanced', [], {'': []}, [], True, True)
-    self._split_help('./pants help --help-advanced', [], {'': []}, [], True, False)
-    self._split_help('./pants help-advanced --help-all', [], {'': []}, [], True, True)
-    self._split_help('./pants compile --help-advanced test', ['compile', 'test'],
-                     {'': [], 'compile': [], 'test': []}, [], True, False)
-    self._split_help('./pants help-advanced compile', ['compile'],
-                     {'': [], 'compile': []}, [], True, False)
-    self._split_help('./pants compile help-all test --help', ['compile', 'test'],
-                     {'': [], 'compile': [], 'test': []}, [], False, True)
-
-  def test_version_request_detection(self):
-    self._split_version_request('./pants -v')
-    self._split_version_request('./pants -V')
-    self._split_version_request('./pants --version')
+    self.assert_valid_split(
+      './pants test.junit --reporting-template-dir=path foo:bar',
+      expected_goals=['test'],
+      expected_scope_to_flags={'': [], 'reporting': ['--template-dir=path'], 'test.junit': []},
+      expected_positional_args=['foo:bar']
+    )
+
+  def test_help_detection(self) -> None:
+    assert_help = partial(
+      self.assert_valid_split, expected_passthru=None, expected_passthru_owner=None, expected_is_help=True,
+    )
+    assert_help_no_arguments = partial(
+      assert_help, expected_goals=[], expected_scope_to_flags={'': []}, expected_positional_args=[],
+    )
```

Comment:
These new helper functions are used for deduplication. Otherwise, everything is the same for this test outside of keyword args.